runtime/compiler/optimizer/JProfilingValue.cpp

```diff
@@ -491,171 +488,127 @@ TR_JProfilingValue::addProfilingTrees(
-
-   /********************* quickInc Block *********************/
-   // If we aren't extending the blocks, anchor stores for the table address and the value
-   TR::Node *quickIncAddress = address;
-   TR::Node *quickIncIndex = index;
-   if (!extendBlocks)
-      {
-      TR::SymbolReference *storedIndexSymRef = NULL;
-      TR::SymbolReference *storedAddressSymRef = NULL;
-      TR::TreeTop::create(comp, quickTest->getEntry(), storeNode(comp, index, storedIndexSymRef));
-      TR::TreeTop::create(comp, quickTest->getEntry(), storeNode(comp, address, storedAddressSymRef));
-      storedAddressSymRef->getSymbol()->setNotCollected();
-
-      quickIncIndex = TR::Node::createLoad(example, storedIndexSymRef);
-      quickIncAddress = TR::Node::createLoad(example, storedAddressSymRef);
-      }
-
-   // Split the block, creating a new block to hold the successful quick increment
-   TR::Block *quickInc = quickTest->split(checkKeyTree->getNextTreeTop(), cfg, true, true);
-   if (extendBlocks)
-      quickInc->setIsExtensionOfPreviousBlock();
-
    if (trace)
       traceMsg(comp, "  Quick increment in block_%d\n", quickInc->getNumber());
 
-   TR::Node *counterOffset = systemConst(example, table->getFreqOffset());
+   TR::Node *counterOffset = systemConst(value, table->getFreqOffset());
    TR::TreeTop *incTree = TR::TreeTop::create(comp, quickInc->getEntry(),
-      incrementMemory(comp, counterType, effectiveAddress(counterType, quickIncAddress, quickIncIndex, counterOffset)));
+      incrementMemory(comp, counterType, effectiveAddress(counterType, address, ternarySelectNode, counterOffset))); 
 
    // Split the block again, after the increment, so cold paths can merge back
-   TR::Block *mainlineReturn = quickInc->split(incTree->getNextTreeTop(), cfg, true, true);
-
-   /********************* slowTest Block *********************/
-   TR::Block *slowTest = TR::Block::createEmptyBlock(comp, MAX_COLD_BLOCK_COUNT + 1);
-   slowTest->setIsCold();
-   lastTreeTop->join(slowTest->getEntry());
-   lastTreeTop = slowTest->getExit();
-   cfg->addNode(slowTest);
-   cfg->addEdge(quickTest, slowTest);
-   cfg->addEdge(slowTest, mainlineReturn);
-   checkKey->setBranchDestination(slowTest->getEntry());
-
-   if (trace)
-      traceMsg(comp, "  Lock test in block_%d\n", slowTest->getNumber());
-
-   // Load the lock and test if its set
-   TR::Node *lockAddress = TR::Node::aconst(example, table->getBaseAddress() + table->getLockOffset());
-   TR::Node *lock = loadValue(comp, lockType, lockAddress);
-   TR::Node *checkTableLock = TR::Node::createif(TR::ifscmplt, lock, TR::Node::sconst(example, 0));
-   TR::TreeTop::create(comp, slowTest->getEntry(), checkTableLock);
-
-   /********************* slowInc Block *********************/
-   TR::Node *slowIncIndex = convertType(lock, systemType, false);
-   if (!extendBlocks)
+   TR::Block *mainlineReturn = quickInc->split(incTree->getNextTreeTop(), cfg, false, true);
```

Comment:
I don't see how you can do fixupCommoning = false and have a merge - you have to uncommon to merge...